### PR TITLE
Add corrections from Fossies extra dictionary

### PR DESCRIPTION
**Pull request by skangas**
This adds most missing corrections from [Fossies extra dictionary](https://fossies.org/codespell_usage_hints.html), with changes made by me. The changes were mostly removing some invalid/duplicate entries, and also adding in many variations, plural forms, etc.

EDIT: See also #2399.

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -4114,6 +4138,7 @@ backedn->backend
 backedns->backends
 backened->backend, blackened,
 backeneds->backends, blackens,
+backets->brackets
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:4141]: `backets->brackets,baskets,buckets,`?

Comment (skangas) [reply codespell_lib/data/dictionary.txt:4141]: Added.

**Comment by skangas** [comment]
Thanks @DimitriPapadopoulos for the review. I've fixed your comment and rebased on top of current master.

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -24717,6 +24968,7 @@ parenthesies->parentheses
 parenthises->parentheses
 parenthsis->parenthesis
 paret->parent, parrot,
+paretheses->parenthesis
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:24971]: `paretheses->parentheses` (plural)

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -26677,6 +26949,7 @@ prohibted->prohibited
 prohibting->prohibiting
 prohibts->prohibits
 proirity->priority
+projcet->project
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:26952]: `projcets->projects`?

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -27128,6 +27409,7 @@ pyramide->pyramid
 pyramides->pyramids
 pyrhon->python
 pyscic->psychic
+pysical->physical
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:27412]: `pysic`, `pysics`?

Note that there's a [pysic](https://thynnine.github.io/pysic/) Python module...

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -30761,6 +31071,7 @@ serailse->serialise
 serailsed->serialised
 serailze->serialize
 serailzed->serialized
+seralized->serialized
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:31074]: `seralize`, `seralizes`, `seralizing`?

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -31674,10 +31996,12 @@ speaces->spaces, species,
 speach->speech
 speacial->special, spacial,
 speacing->spacing
+spearated->separated
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:31999]: `spearate`, `spearates`, `spearating`?

File: codespell_lib/data/dictionary.txt [hunk 7]
```diff
@@ -34091,13 +34451,16 @@ thann->than, thank,
...
 thant->than
+thar->than, that,
 thare->there
 thast->that, that's,
 that;s->that's
 thatn->that, than,
 thats'->that's
 thats->that's
 thats;->that's
+thaught->thought
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:34462]: `thaught->thought, taught,`?
`thaughts->thoughts, taughts,`?

File: codespell_lib/data/dictionary.txt [hunk 8]
```diff
@@ -34649,6 +35016,7 @@ tranport->transport
 tranported->transported
 tranporting->transporting
 tranports->transports
+tranpose->transpose
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:35019]: `tranposed`?

File: codespell_lib/data/dictionary.txt [hunk 9]
```diff
@@ -5268,7 +5307,7 @@ callbck->callback
 callcack->callback
 callcain->callchain
 calld->called
-calle->called
+calle->called, caller,
 callef->called
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:5317]: Fixed.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:24971]: Well spotted! Fixed.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:26952]: Added.

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:27412]: Added. I moved `pysic` to `dictionary_code.txt` for good measure though. I also added `pysically`.

File: codespell_lib/data/dictionary.txt (hunk 5 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:31074]: Added.

File: codespell_lib/data/dictionary.txt (hunk 6 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:31999]: Added.

File: codespell_lib/data/dictionary.txt (hunk 7 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:34462]: > `thaught->thought, taught,`?

Added.

> `thaughts->thoughts, taughts,`?

I'm not sure about `taughts`. Is it a correct word?

File: codespell_lib/data/dictionary.txt (hunk 8 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:35019]: Added. Also added `tranposes`, `tranposing`.

File: codespell_lib/data/dictionary.txt (hunk 7 above)
Comment (DimitriPapadopoulos) [reply codespell_lib/data/dictionary.txt]: No :blush:

**Review by DimitriPapadopoulos — APPROVED** [review]

File: codespell_lib/data/dictionary.txt (hunk 7 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt]: > No :blush: 

:smile:  :+1:

**Comment by skangas** [comment]
Thanks for the quick merge!

**Pull request merged**